src/record/record-handler.js

```diff
@@ -241,26 +247,57 @@ RecordHandler.prototype._createOrRead = function (socketWrapper, message) {
   )
 }
 
+RecordHandler.prototype._createAndUpdate = function (socketWrapper, message) {
+  const recordName = message.data[0]
+  const isPatch = message.data.length === 5
+  message.action = isPatch ? C.ACTIONS.PATCH : C.ACTIONS.UPDATE
+  const onComplete = function (record) {
```

Comment:
be interesting to run a simple benchmark to see if this closure being internal actually results in a compiler overhead